### PR TITLE
DDCE-4709 Upgrade RAS to Upscan

### PR DESCRIPTION
**Pull request by zmdnoor**

### CODE REVIEW

File: app/controllers/UpscanController.scala [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                                     val mcc: MessagesControllerComponents,
+                                     implicit val appConfig: ApplicationConfig,
+                                     fileUploadView: views.html.file_upload,
+                                     fileUploadSuccessView: views.html.file_upload_successful,
+                                     cannotUploadAnotherView: views.html.cannot_upload_another_file
+																		) extends FrontendController(mcc) with RasController with PageFlowController with Logging {
+
+	implicit val ec: ExecutionContext = mcc.executionContext
+
+  def get: Action[AnyContent] = Action.async {
```
Comment (MRBeresford) [inline app/controllers/UpscanController.scala:45]: this is a very large method, honestly reading it I got quite confused. I think we should pull this method apart a bit and make some smaller methods just to make it more readable. I had a quick go let me know what you think...

File: app/controllers/UpscanController.scala [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+      case Right(userId) =>
+        sessionService.fetchRasSession().flatMap {
+          case Some(session) =>
+            session.file match {
+              case Some(file) =>
+                filesSessionService.createFileSession(userId,file.id).map {
+                  case true =>
+                    logger.info(s"[UpscanController][uploadSuccess] upload has been successful for userId ($userId)")
+                    Ok(fileUploadSuccessView())
+                  case _ =>
```
[the remark this answers is not in the document]
Comment (MRBeresford) [reply app/controllers/UpscanController.scala:135]: this applies to the rest of this file as well 👍

**Review by MRBeresford — CHANGES_REQUESTED** [review]
Initial review, I havnt run any tests yet just looked through the code and left some comments. Let me know what you think!

File: app/controllers/UpscanController.scala (hunk 1 above)
Comment (zmdnoor) [reply app/controllers/UpscanController.scala]: See latest changes

**Comment by zmdnoor** [comment]
Changes implemented, have a look and let me know :)

**Review by MRBeresford — CHANGES_REQUESTED** [review]
More comments, enjoy! 😈

**Review by MRBeresford — APPROVED** [review]
LGTM

**Pull request merged**